apollo-router/src/graphql/response.rs

```diff
@@ -3,18 +3,25 @@ use std::time::Instant;
 
 use apollo_compiler::response::ExecutionResponse;
 use bytes::Bytes;
+use displaydoc::Display;
 use serde::Deserialize;
 use serde::Serialize;
 use serde_json_bytes::ByteString;
 use serde_json_bytes::Map;
 
 use crate::error::Error;
-use crate::error::FetchError;
 use crate::graphql::IntoGraphQLErrors;
 use crate::json_ext::Object;
 use crate::json_ext::Path;
 use crate::json_ext::Value;
 
+#[derive(thiserror::Error, Display, Debug, Eq, PartialEq)]
+#[error("GraphQL response was malformed: {reason}")]
+pub(crate) struct MalformedResponseError {
+    /// The reason the deserialization failed.
+    pub(crate) reason: String,
+}
```

Comment:
Previously, we reported all errors related to parsing/validation graphql response as `FetchError::SubrequestMalformedResponse` which made this code only usable for subgraph responses. Since now we have generic error it could be reused for coprocessor responses.